services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/controllers/NotificationController.java

```diff
@@ -0,0 +1,583 @@
+import us.dot.its.jpo.ode.api.accessors.notifications.StopLineStopNotification.StopLineStopNotificationRepository;
+import us.dot.its.jpo.ode.api.accessors.notifications.TimeChangeDetailsNotification.TimeChangeDetailsNotificationRepository;
+import us.dot.its.jpo.ode.mockdata.MockNotificationGenerator;
+
+@Slf4j
+@RestController
+@ConditionalOnProperty(name = "enable.api", havingValue = "true", matchIfMissing = false)
+public class NotificationController {
+
+    @Autowired
+    IntersectionReferenceAlignmentNotificationRepository intersectionReferenceAlignmentNotificationRepo;
+
+    @Autowired
+    LaneDirectionOfTravelNotificationRepository laneDirectionOfTravelNotificationRepo;
+
+    @Autowired
+    MapBroadcastRateNotificationRepository mapBroadcastRateNotificationRepo;
+
+    @Autowired
+    SignalGroupAlignmentNotificationRepository signalGroupAlignmentNotificationRepo;
+
+    @Autowired
+    SignalStateConflictNotificationRepository signalStateConflictNotificationRepo;
+
+    @Autowired
+    SpatBroadcastRateNotificationRepository spatBroadcastRateNotificationRepo;
+
+    @Autowired
+    ConnectionOfTravelNotificationRepository connectionOfTravelNotificationRepo;
+
+    @Autowired
+    TimeChangeDetailsNotificationRepository timeChangeDetailsNotificationRepo;
+
+    @Autowired
+    StopLineStopNotificationRepository stopLineStopNotificationRepo;
+
+    @Autowired
+    StopLinePassageNotificationRepository stopLinePassageNotificationRepo;
+
+    @Autowired
+    ActiveNotificationRepository activeNotificationRepo;
+    @Autowired
+    ConflictMonitorApiProperties props;
+
+    @RequestMapping(value = "/notifications/active", method = RequestMethod.GET, produces = "application/json")
+    @PreAuthorize("@PermissionService.isSuperUser() || (@PermissionService.hasIntersection(#intersectionID) and @PermissionService.hasRole('USER')) ")
+    public ResponseEntity<List<Notification>> findActiveNotification(
+            @RequestParam(name = "intersection_id", required = false) Integer intersectionID,
+            @RequestParam(name = "road_regulator_id", required = false) Integer roadRegulatorID,
+            @RequestParam(name = "notification_type", required = false) String notificationType,
+            @RequestParam(name = "key", required = false) String key,
+            @RequestParam(name = "test", required = false, defaultValue = "false") boolean testData) {
+        if (testData) {
+            List<Notification> list = new ArrayList<>();
+            list.add(MockNotificationGenerator.getConnectionOfTravelNotification());
+            return ResponseEntity.ok(list);
+        } else {
+            Query query = activeNotificationRepo.getQuery(intersectionID, roadRegulatorID, notificationType, key);
+            long count = activeNotificationRepo.getQueryResultCount(query);
+            if (count <= props.getMaximumResponseSize()) {
```

Comment:
suggestion: could we achieve better functionality by implementing `limit()` on the query instead of hitting the db twice for count and find?

Reply:
several other methods in here have the same logic that could be tweaked